services/web/server/tests/unit/isolated/test_application_settings.py

```diff
@@ -2,44 +2,214 @@
+    #         - WEBSERVER_LOGLEVEL=${LOG_LEVEL:-WARNING}
+    #     env_file:
+    #         - ../.env
+
+    monkeypatch_setenv_default("CATALOG_HOST", "catalog")
+    monkeypatch_setenv_default("CATALOG_PORT", "8000")
+    monkeypatch.setenv("DIAGNOSTICS_MAX_AVG_LATENCY", "30")
+    monkeypatch_setenv_default("DIRECTOR_HOST", "director")
+    monkeypatch_setenv_default("DIRECTOR_PORT", "8080")
+    monkeypatch_setenv_default("DIRECTOR_V2_HOST", "director-v2")
+    monkeypatch_setenv_default("DIRECTOR_V2_PORT", "8000")
+    monkeypatch_setenv_default("STORAGE_HOST", "storage")
+    monkeypatch_setenv_default("STORAGE_PORT", "8080")
+    monkeypatch_setenv_default("SWARM_STACK_NAME", "simcore")
+    monkeypatch.setenv("WEBSERVER_LOGLEVEL", os.environ.get("LOG_LEVEL", "WARNING"))
+
+
+@pytest.fixture
+def app_settings(mock_webserver_service_environment) -> ApplicationSettings:
+
+    app = web.Application()
 
     # init and validation happens here
-    setup_settings(app)
+    settings = setup_settings(app)
+    print("app settings:\n", settings.json(indent=1))
 
     assert APP_SETTINGS_KEY in app
-    return app[APP_SETTINGS_KEY]
+    assert app[APP_SETTINGS_KEY] == settings
+    return settings
 
 
-def test_settings_constructs(settings: ApplicationSettings):
-    assert "vcs_url" in settings.public_dict()
+@pytest.fixture(
+    params=[
+        "server-docker-prod.yaml",
+    ]
+)
+def app_config(request, mock_webserver_service_environment) -> Dict:
+    parser = setup_parser(argparse.ArgumentParser("test-parser"))
+    config = parse(["-c", request.param], parser)
+    print("app config [request.param]:\n", json.dumps(config, indent=1))
+    return config
+
+
+# TESTS -----------------------------
+
+
+def test_app_settings_with_prod_config(
+    app_config: Dict, app_settings: ApplicationSettings
+):
+    # Ensures all plugins are enabled for this test
+    assert app_settings.WEBSERVER_EMAIL is not None
+    assert app_settings.WEBSERVER_ACTIVITY is not None
+    assert app_settings.WEBSERVER_REDIS is not None
+    assert app_settings.WEBSERVER_TRACING is not None
+    assert app_settings.WEBSERVER_CATALOG is not None
+    assert app_settings.WEBSERVER_DIRECTOR is not None
+    assert app_settings.WEBSERVER_STORAGE is not None
+    assert app_settings.WEBSERVER_DIRECTOR_V2 is not None
+    assert app_settings.WEBSERVER_RESOURCE_MANAGER is not None
+    assert app_settings.WEBSERVER_LOGIN is not None
+
+    # This is basically how the fields in ApplicationSettings map the trafaret's config file
+    #
+    # This test compares the config produced by trafaret against
+    # the equilalent fields captured by ApplicationSettings
+    #
+    # This guarantees that all configuration from the previous
+    # version can be recovered with the new settings approach
+    #
+    # This test has been used to guide the design of new settings
+    #
+
+    # NOTE: new settings truncate to 32
```

Comment:
I'm not sure I understand why, sorry.

Reply:
Basically the secret key has to be 32 bytes long. The new settings truncates the envvar to this value. SEE 
https://github.com/pyca/cryptography/blob/main/src/cryptography/fernet.py#L38

I will improve doc

Reply:
I finally changed the session settings and added extra tests to make sure it does not break when encryption happens